src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -225,15 +228,18 @@ private void VerifyAdditionalFile(AdditionalText file)
         /// <summary>
         /// Returns diagnostics produced by all <see cref="Analyzers"/>.
         /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsAsync()
         {
-            return GetAnalyzerDiagnosticsAsync(_cancellationToken);
+            return GetAnalyzerDiagnosticsAsync(cancellationToken: default);
         }
 
         /// <summary>
         /// Returns diagnostics produced by all <see cref="Analyzers"/>.
         /// </summary>
-        public async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsAsync(CancellationToken cancellationToken)
+#pragma warning disable RS0027 // API with optional parameter(s) should have the most parameters amongst its public overloads
+        public async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsAsync(CancellationToken cancellationToken = default)
```

Comment:
❗ This isn't going to work as expected

Reply:
Suggestion: make `cancellationToken` here required, and allow both methods to be public

Reply:
> The compiler will still bind to the parameterless method if the cancellation token is not provided here.

Yes.  but that's fine.  That's why it's not obsoleted.